Add documentation example for internal links with `set_link()` to pages with variable page numbers

biermanr opened the pull request:
<!--
Thanks for your interest in the project. Bugs filed and PRs submitted are appreciated!

Some guidelines are provided like this, in HTML comments, to expedite the code review before merging your contribution.

First, please make sure that you have read the documentation page on fpdf2 development:
https://py-pdf.github.io/fpdf2/Development.html

If you're new to contributing to open source projects,
you might find this free video course helpful: http://kcd.im/pull-request
-->

<!-- What changes are being made? (What feature/bug is being fixed here?) -->

Add example to documentation of internal links showing how to use `set_link()` to change the referenced
page of a previous `add_link()` command. Same topic as discussion from https://github.com/py-pdf/fpdf2/discussions/729

<!-- Have you done all of these things?  -->
**Checklist**:

<!-- To check an item, place an "x" in the box like so: "- [x] Item description"
     Add "N/A" to the end of each line that's irrelevant to your changes -->

- [ ] A unit test is covering the code added / modified by this PR
(no test added, `test/test_links.py: test_later_call_to_set_link` covers use case described in added docs)

- [x] This PR is ready to be merged <!-- In your opinion, can this be merged as soon as it's reviewed? Else, this can be turned into a Draft PR -->

- [ ] In case of a new feature, docstrings have been added, with also some documentation in the `docs/` folder
(no new feature)

- [x] A mention of the change is present in `CHANGELOG.md`

<!-- Feel free to add additional comments, and to ask questions if some of those guidelines are unclear to you! -->

<!--
Once a PR is merged, maintainers will add your name to the contributors table in README.md.
If they forget, or you do not wish this to happen, please mention it.
-->

By submitting this pull request, I confirm that my contribution is made under the terms of the [GNU LGPL 3.0 license](https://github.com/py-pdf/fpdf2/blob/master/LICENSE).

Lucas-C commented on CHANGELOG.md:
```diff
@@ -22,6 +22,7 @@ This can also be enabled programmatically with `warnings.simplefilter('default',
 
 ## [2.8.3] - 2025-04-22
 ### Added
+* documentation on [internal linking with variable page numbers](https://py-pdf.github.io/fpdf2/Links.html#internal-links)
```
Could you please move this to the `[2.8.4] - Not released yet` section above?

Lucas-C commented on docs/Links.md:
````diff
@@ -98,6 +98,53 @@ pdf.cell(text="Internal link to first page", border=1, link=link)
+
+pdf.output("internal_link_unknown_pages.pdf")
+```
+
+
+
````
```suggestion

```

Lucas-C reviewed (approved):
Thank you for this nice addition to our documentation @biermanr 👍

Lucas-C commented:
@allcontributors please add @biermanr for doc

biermanr commented:
Hi @Lucas-C, thank you for your help on this, I've accepted your whitespace suggestion and moved the CHANGELOG.md line

I've also added a new test in `test/test_links.py` and created a corresponding PDF

biermanr commented:
@Lucas-C sorry you're definitely correct, I overlooked the existing `test_later_call_to_set_link` test which already covers the internal linking use case I added to the documentation.

I've removed the duplicate test I added, and removed the added "test pdf" so now we're back to just the docs and the CHANGELOG edits

Lucas-C commented:
Thank you @biermanr 👍

Merged
